mesh/meshdb.go

```diff
@@ -382,38 +382,12 @@ func getTransactionDestKeyPrefix(l types.LayerID, account types.Address) []byte
 	return []byte(str)
 }
 
-type dbTransaction struct {
-	*types.Transaction
-	Origin types.Address
-}
-
-func newDbTransaction(tx *types.Transaction) *dbTransaction {
-	return &dbTransaction{Transaction: tx, Origin: tx.Origin()}
-}
-
-func (t dbTransaction) getTransaction() *types.Transaction {
-	t.Transaction.SetOrigin(t.Origin)
-	return t.Transaction
-}
-
-func (m *DB) writeTransactions(l types.LayerID, txs []*types.Transaction) error {
+func (m *DB) writeTransactions(l types.LayerID, txs []types.Transaction) error {
 	batch := m.transactions.NewBatch()
 	for _, t := range txs {
-		bytes, err := types.InterfaceToBytes(newDbTransaction(t))
-		if err != nil {
-			return fmt.Errorf("could not marshall tx %v to bytes: %v", t.ID().ShortString(), err)
-		}
-		if err := batch.Put(t.ID().Bytes(), bytes); err != nil {
-			return fmt.Errorf("could not write tx %v to database: %v", t.ID().ShortString(), err)
-		}
-		// write extra index for querying txs by account
-		if err := batch.Put(getTransactionOriginKey(l, t), t.ID().Bytes()); err != nil {
-			return fmt.Errorf("could not write tx %v to database: %v", t.ID().ShortString(), err)
-		}
-		if err := batch.Put(getTransactionDestKey(l, t), t.ID().Bytes()); err != nil {
-			return fmt.Errorf("could not write tx %v to database: %v", t.ID().ShortString(), err)
+		if err := m.WriteTransaction(l, t); err != nil {
+			return err
 		}
-		m.Debug("wrote tx %v to db", t.ID().ShortString())
 	}
 	err := batch.Write()
```

Comment:
This method originally wrote transactions atomically, if the batch failed, nothing was written.

I'm not 100% sure we fully implemented the logic for what happens on failure, but this change feels a little like a step backwards. Worth reconsidering.

Reply:
Fixed
https://github.com/spacemeshos/go-spacemesh/blob/80b86b1c9f9f93da13bcca40d663417ccec48603/mesh/meshdb.go#L385